src/compiler/transformers/classFields.ts

```diff
@@ -738,7 +738,7 @@ namespace ts {
                     visitNode(node.right, visitor)
                 );
                 const expr = some(pendingExpressions) ?
-                    factory.inlineExpressions(compact([...pendingExpressions!, node])) :
+                    factory.inlineExpressions(compact([...pendingExpressions, node])) :

```

Comment:
This one is actually unfortunate - `pendingExpressions` is actually typed as `never` because we have an earlier assignment of `undefined` followed by a check for `some(...)` which is a type predicate on `readonly T[]`. Since you can't narrow down a `readonly T[]` from an `undefined`, it leaves us with `never`.

The real culprit is #9998, but it is also strange that we just allow a spread of type `never`.